src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        joinedString += '';
+      } else {
+        joinedString += this[i];
+      }
+
+      if (i !== this.length - 1) {
+        joinedString += String(separator);
+      }
+    }
+
+    return joinedString;
   };
 }
 
+const source = [0, 1, 2, 3];
+
+applyCustomJoin();
+
+source.join2(null);
+
```

Comment:
do we need this part for tests? 

(P.S. in a real project, we would probably write this data, but in the case of tasks, we will receive several different variations instead of this single option)